cmd/internal/ecs/add.go

```diff
@@ -870,6 +879,121 @@ func modifyTaskState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowS
+	// that leaving them null will leave them unchanged.
+	input := &ecs.UpdateServiceInput{
+		Service:        wf.ecsServiceARN.Use(), // documentation says "name"?
+		Cluster:        wf.ecsClusterARN.Use(),
+		TaskDefinition: wf.ecsTaskDefinitionARN.Use(),
+	}
+	_, err = wf.ecsClient.UpdateService(wf.ctx, input)
+	if err != nil {
+		if uoe, unauth := isUnauthorized(err); unauth {
+			printer.Errorf("The provided credentials do not have permission to update the ECS service %q (operation %s).\n",
+				wf.ecsServiceARN, uoe.OperationName)
+			return awf_error(errors.New("Failed to update the ECS service due to insufficient permissions."))
+		}
+	}
+	printer.Infof("Updated service %q with new version of task definition.\n", wf.ecsService)
+
+	// Try to tag the service; this can't be done in the UpdateService call
+	// but its failure is non-fatal.
+	tagInput := &ecs.TagResourceInput{
+		ResourceArn: wf.ecsServiceARN.Use(),
+		Tags: []types.Tag{{
+			Key:   aws.String(akitaModificationTagKey),
+			Value: aws.String(akitaModificationTagValue),
+		}},
+	}
+	_, tagErr := wf.ecsClient.TagResource(wf.ctx, tagInput)
+	if tagErr == nil {
+		printer.Infof("Tagged service %q with %q.\n", wf.ecsService, akitaModificationTagKey)
+	} else {
+		if uoe, unauth := isUnauthorized(tagErr); unauth {
+			printer.Warningf("The provided credentials do not have permission to tag the ECS service %q (operation %s).\n",
+				wf.ecsServiceARN, uoe.OperationName)
+		} else {
+			printer.Warningf("Failed to tag the ECS service: %v\n", tagErr)
+		}
+		printer.Infof("The service has been modified, but it will be harder to locate it to roll back changes.\n")
+	}
+
+	return awf_next(waitForRestartState)
+}
+
+func waitForRestartState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error) {
+	reportStep("Wait for ECS Service")
+	startTime := time.Now()
+	endTime := startTime.Add(5 * time.Minute)
+
+	failure := errors.New("Cannot determine whether service was successfully upgraded. Please check the AWS console.\n")
+
+	// Check that there's a new deployment in the service
+	deploymentID, deployment, err := wf.GetDeploymentMatchingTask(wf.ecsServiceARN)
+	if err != nil {
+		if errors.Is(err, noDeploymentFound) {
+			printer.Errorf("Could not locate a deployment for the new task definition \"%s:%s\" (%s).\n",
+				wf.ecsTaskDefinitionFamily,
+				wf.ecsTaskDefinition.Revision,
+				wf.ecsTaskDefinitionARN)
+		} else {
+			printer.Errorf("Error checking service state: %v\n", err)
+		}
+		return awf_error(failure)
+	}
+	printer.Infof("Found new deployment with ID %q; its state is %s.\n", deploymentID, deployment.RolloutState)
+
+	if deployment.RolloutState == types.DeploymentRolloutStateInProgress {
+		printer.Infof("Waiting for the deployment to reach COMPLETED.\n")
+	}
+
+	ticker := time.NewTicker(15 * time.Second)
+	for deployment.RolloutState == types.DeploymentRolloutStateInProgress {
```

Comment:
I think there's a race condition where `deployment.RolloutState == types.DeploymentRolloutStateFailed` before reaching this loop, in which case the CLI will report success despite the failure.

Maybe break on line 985 and check/report on failure after the loop?

Reply:
Good catch, thanks.  I don't think we need to break, the loop condition will terminate.

Reply:
in https://github.com/akitasoftware/akita-cli/pull/186/commits/e67a85920c734666661216bc194f17033a593ab5